### DISCUSSION TITLE
series analysis threshold

### DISCUSSION THREAD

**Discussion by ghost** [dtcenter/METplus#1662] [Configuration] [answered]
Hi,
I am trying to make the attached scripts work for Series Analysis and it shows error like 
ERROR: Cannot set FCST_VAR1_THRESH or OBS_VAR1_THRESH if BOTH_VAR1_THRESH is set.
06/14 17:37:15Z run-METplus-metplus: ERROR:  Cannot set FCST_VAR1_THRESH or OBS_VAR1_THRESH if BOTH_VAR1_THRESH is set

Any help on this is appreciated. 

[grid_stat_icap_Taod_series.conf.txt](https://github.com/dtcenter/METplus/files/8902169/grid_stat_icap_Taod_series.conf.txt)
[run.txt](https://github.com/dtcenter/METplus/files/8902202/run.txt)

**Comment by ghost** [comment]
even with specified thresholds it shows same error. Is that some other way needed to point metplus scripts for series analysis ?

**Comment by jprestop** [comment, marked as answer]
Hi @perthsb.  Thank you for your question.

I see that you are running:
```
run_metplus.py \
${MET_PLUS_CONF}/SeriesAnalysis.conf \
${METPLUS_TUTORIAL_DIR}/grid_stat_icap_Taod_series.conf \
config.OUTPUT_BASE=${METPLUS_TUTORIAL_DIR}/output_series
```
I see that in the default use case file /gpfs/dell2/emc/verification/noscrub/emc.metplus/METplus/METplus-4.0.0/parm/use_cases/met_tool_wrapper/SeriesAnalysis/SeriesAnalysis.conf  the following is set:
```
BOTH_VAR1_THRESH = gt12.7, gt25.4, gt50.8, gt76.2
```
And, in your user config file /gpfs/dell2/emc/verification/noscrub/Partha.Bhattacharjee/Aerosol/GSD_Chem/METplus_test/grid_stat_icap_Taod_series.conf, the following is set:
```
FCST_VAR1_THRESH =
OBS_VAR1_THRESH =
```

The code interprets this as having both FCST_VAR1_THRESH/OBS_VAR1_THRESH and BOTH_VAR1_THRESH being set, hence the error you are receiving.  

You would need to comment out
```
FCST_VAR1_THRESH =
OBS_VAR1_THRESH =
```
in your grid_stat_icap_Taod_series.conf if you wanted to use the default values of 
```
BOTH_VAR1_THRESH = gt12.7, gt25.4, gt50.8, gt76.2
```

OR

If instead you wanted no thresholds set, you would need to override “BOTH_VAR1_THRESH” in your 
grid_stat_icap_Taod_series.conf file by setting:
```
BOTH_VAR1_THRESH = 
```
and commenting out 
```
FCST_VAR1_THRESH =
OBS_VAR1_THRESH =
```

Does that make sense?  

The logic could be improved to consider the empty values as unset.   We believe the motivation of the requirement came from users doing exactly what they are doing -- reading in two config files that have different field info set in each. If the code guessed which one to use, it could silently use a mix of both config file settings and be even more confusing.  What are your thoughts on this?

**Comment by ghost** [comment]
Thanks @jprestop . So when I run GridStat using same dataset, I set both 
FCST_VAR1_THRESH and OBS_VAR1_THRESH as  >=0.1, >=0.2, >=0.4 and it run without any problem. So it seems that for series_analysis it is not sufficient, I need to set BOTH_VAR1_THRESH = separately here as I am pointing to different conf BASE File. 
That brings to another question, is it possible to run series_analysis using 
/gpfs/dell2/emc/verification/noscrub/emc.metplus/METplus/METplus-4.0.0/parm/metplus_config/defaults.conf  or I have to point to

/gpfs/dell2/emc/verification/noscrub/emc.metplus/METplus/METplus4.0.0/parm/use_cases/met_tool_wrapper/SeriesAnalysis/SeriesAnalysis.conf and make changes in my .conf file accordingly ?

**Reply by jprestop** [reply]
To confirm, when you run GridStat using the same dataset AND you are using the file [parm/use_cases/met_tool_wrapper/GridStat/GridStat.conf](https://github.com/dtcenter/METplus/blob/main_v4.1/parm/use_cases/met_tool_wrapper/GridStat/GridStat.conf) as your default file, it would be the case that it would run without any problem because I see that BOTH_VAR1_THRESH is not set in the GridStat.conf file. 

With METplus, you have a lot of control over your configuration.  There is no need to use the SeriesAnalysis.conf file, but it is a starting point for users to ensure that all of the necessary values are set.  You could run simply with your user config file (defaults.conf will be automatically used, i.e. it is not necessary to put that in your command line invocation of run_metplus.py), however, I would suggest looking over the settings in SeriesAnalysis.conf to ensure that you have everything you need to run Series-Analysis.  So, if you did decide to use only your user config file, you may want to copy and paste from the SeriesAnalysis.conf file as needed in order to ensure you have the necessary variables set.  

I hope that helps.  Please let us know if you have additional questions. Or, if you feel we have answered your questions, please select the "Mark as answer" button appropriately. Once we see that you've done that, we'll "Lock conversation" on this discussion to prevent future posts. That's how we encourage users to create new discussions for new questions.

**Reply by ghost** [reply]
thanks @jprestop . I tried using parm/use_cases/met_tool_wrapper/GridStat/GridStat.conf to run Series analysis but it did not run. output log  shows it needs some "series_analysis" specific settings in .conf file. Is it because I set process = series_analysis so it looking for that paricular setting ? Attached is the .conf and outputlog for you to quick look into it.
[grid_stat_icap_Taod_series.conf.txt](https://github.com/dtcenter/METplus/files/8911933/grid_stat_icap_Taod_series.conf.txt)
[output.txt](https://github.com/dtcenter/METplus/files/8911934/output.txt)

**Reply by jprestop** [reply]
GridStat.conf is meant for Grid-Stat runs and SeriesAnalysis.conf is meant for Series-Analysis runs.  I would not suggest using GridStat.conf as a starting point for running Series-Analysis.  Please take a look at the fields set in SeriesAnalysis.conf in putting together your config file, give it a try, and let us know if you need any further guidance.

**Reply by jprestop** [reply]
Hi @perthsb.  I just wanted to follow up and see if you were able to resolve the problem you were experiencing?

**Reply by ghost** [reply]
Hi @jprestop , I have not tried yet, but you may close this ticket. I will open another one , if I encounter issues.
Thanks

**Reply by jprestop** [reply]
Ok, thank you @perthsb.
